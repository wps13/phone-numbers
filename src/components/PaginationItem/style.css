.pagination-item {
  padding: 0.5em 1em;
  font-weight: 900;
  border: 1px solid #000;
  border-radius: 5px;
  text-align: center;
  display: inline-block;
  margin: 0 10px;
}

.pagination-item.number {
  width: 40px;
  display: none;
}

.pagination-item.text {
  width: 80px;
}

@media (min-width: 768px) {
  .pagination-item.number {
    display: inline-block;
  }
}
