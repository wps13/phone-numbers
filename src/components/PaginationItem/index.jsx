import React from "react";

import "./style.css";

const PaginationItem = ({ text, type }) => {
  return (
    <div className={"pagination-item " + type} data-testid="pagination-item">
      <p>{text}</p>
    </div>
  );
};

export default PaginationItem;
