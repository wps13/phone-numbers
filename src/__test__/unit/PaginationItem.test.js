import React from "react";

import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import PaginationItem from "../../components/PaginationItem";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<PaginationItem />, div);
});

it("render number correctly", () => {
  const { getByTestId } = render(<PaginationItem text={1} type="number" />);
  expect(getByTestId("pagination-item")).toHaveTextContent("1");
});

it("render text correctly", () => {
  const { getByTestId } = render(<PaginationItem text="Next" type="text" />);
  expect(getByTestId("pagination-item")).toHaveTextContent("Next");
});

it("matches snapshot", () => {
  const tree = renderer.create(<PaginationItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
