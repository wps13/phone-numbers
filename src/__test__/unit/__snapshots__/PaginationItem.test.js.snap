// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="pagination-item undefined"
  data-testid="pagination-item"
>
  <p />
</div>
`;
